### PR TITLE
[mergebot] refactor dispatching logic

### PR DESCRIPTION
**Pull request by suo**
Small refactor; `dispatchEvent` had many mutually exclusive arguments
that we unrolled with control flow inside the function. This is an
indication that that the function is factored incorrectly.

This PR lifts all the branching logic out of the function and keeps only
the common bits.

Also, refactor our tests to be more explicit (matching objects instead of string matching).

### CODE REVIEW

File: torchci/lib/bot/mergeBot.ts [hunk 1]
```diff
@@ -138,14 +104,23 @@ function mergeBot(app: Probot): void {
...
-        return await handleRevert(args.message);
+        return await dispatchEvent("try-revert", {
+          reason: args.message,
+        });
       case "merge":
-        return await handleMerge(args.force, args.green, args.all_green);
+        return await dispatchEvent("try-merge", {
+          force: args.force,
+          on_green: args.green,
+          all_green: args.all_green,
```
Comment (clee2000) [inline torchci/lib/bot/mergeBot.ts:114]: i think this will require changes in trymerge.yml b/c it uses -n right now?

Comment (suo) [reply torchci/lib/bot/mergeBot.ts:114]: oh darn

**Review by seemethere — APPROVED** [review]

**Review by zengk95 — APPROVED** [review]

File: torchci/lib/bot/mergeBot.ts [hunk 2]
```diff
@@ -93,20 +58,21 @@ function mergeBot(app: Probot): void {
...
       } else {
         await addComment(
           ctx,
           "You don't have permissions to rebase this PR, only the PR author and pytorch organization members may rebase this PR."
         );
       }
     }
-    if (ctx.payload.comment.user.id == 54816060) {
+
+    if (ctx.payload.comment.user.id == PYTORCH_BOT_USER_ID) {
```
Comment (malfet) [inline torchci/lib/bot/mergeBot.ts:70]: @seemethere refactors this code in #380

**Review by malfet — APPROVED** [review]

File: torchci/lib/bot/mergeBot.ts (hunk 1 above)
Comment (malfet) [reply torchci/lib/bot/mergeBot.ts:114]: Agree, it does not validate truthiness of the statement, just whether option is present or not. But one can probably add function that yanks false arguments

**Pull request closed without merging**